pkg/segment/writer/segwriter.go

```diff
@@ -445,7 +443,7 @@ func getSegStore(streamid string, ts_millis uint64, table string, orgId uint64)
 			return nil, fmt.Errorf("getSegStore: max allowed segstores reached (%d)", maxAllowedSegStores)
 		}
 
-		segstore = &SegStore{Lock: sync.Mutex{}, OrgId: orgId, firstTime: true}
+		segstore = &SegStore{Lock: sync.Mutex{}, OrgId: orgId, firstTime: true, pqMatches: make(map[string]*pqmr.PQMatchResults), LastSegPqids: make(map[string]struct{})}
```

Comment:
nit: now that this has gotten a bit longer, it's probably worth putting each field on a separate line

Reply:
addressed in the latest commit.